bin/faucet/src/faucet/mod.rs

```diff
@@ -38,6 +40,54 @@ mod store;
 
 pub const DISTRIBUTE_FUNGIBLE_ASSET_SCRIPT: &str = include_str!("distribute_fungible_asset.masm");
 
+// FAUCET PROVER
+// ================================================================================================
+
+/// Represents a transaction prover which can be either local or remote, and is used to prove
+/// transactions minted by the faucet.
+enum FaucetProver {
+    Local(LocalTransactionProver),
+    Remote(RemoteTransactionProver),
+}
+
+impl FaucetProver {
+    /// Creates a new local prover.
+    ///
+    /// It uses the default proving options.
+    fn local() -> Self {
+        Self::Local(LocalTransactionProver::new(ProvingOptions::default()))
+    }
+
+    /// Creates a new remote prover.
+    ///
+    /// # Arguments
+    ///
+    /// * `endpoint` - The endpoint to connect to the remote prover.
+    fn remote(endpoint: impl Into<String>) -> Self {
```

Comment:
nit: Tend to prefer to carry the type-safety as deep into the stack as possible. So would expect this fn to take a `Url` which has been parsed by config/flags prior to this call.

If that's impractical for any reason here feel free to resolve this comment.